app/jobs/distribution_reminder_job.rb

```diff
@@ -1,9 +1,11 @@
 # This job notifies a Partner that they have a distribution scheduled to be sent in 24 hours
-class DistributionReminderJob
-  include Sidekiq::Worker
-
+class DistributionReminderJob < ApplicationJob
   def perform(dist_id)
     distribution = Distribution.find(dist_id)
+
+    # NOTE: This is being also checked in the DistributionMailer itself
+    return if distribution.past? || !distribution.partner.send_reminders
```

Comment:
This fixes an inefficiency we identified -- jobs were getting queued up that would be DOA. This pre-empts those from being added at all.

Reply:
Good catch!
![](https://media1.giphy.com/media/aCKMaeduKfFXG/giphy.gif)